### ISSUE TITLE
Misaligned Gizmos/Movement Arrows

### ISSUE THREAD

**Issue by ToxicHolyGrenade** [Winded/RagdollMover#27]
Another problem is that through some reason when saving the game and going back to working on the ragdolls, the gizmos/movement arrows will be out of sync, and also misaligned (i.e. the arrows not pointing in the right directions, like instead of being straight in line, the arrows point in different angles, which should not be happening).

There are two theories on why this occurs. Either A), it happens with all Gmod saves, which is something that can be concerning for not just Gmod artists, but also Gmod animators. B), is with the ragdoll duplicates.

Can this be classified as a "will-not-fix" kind of issue, or can this be patched out?

(For the record, I am getting this problem more recognition as possible, because I do emphasize that this is a problem that has persisted for goodness knows how long.)

**Comment by penolakushari** [comment]
I've never had this issue with current wip ragdoll mover or the workshop one. As I did ask on steam, did you happen to disable all your addons except for ragdoll mover and check with that?

**Comment by ToxicHolyGrenade** [comment]
@penolakushari From what I had seen before, nothing is coming up on my Garry's Mod error notices that there are conflicting addons regarding Ragdoll Mover.

**Comment by ToxicHolyGrenade** [comment]
> Still do a run without any other addons enabled just to be sure

Okay, I tried to recreate the same problem but with the rest of the addons being disabled, and sadly, the end result is the same. So it has nothing to do with conflicting addons.

**Comment by ToxicHolyGrenade** [comment]
@penolakushari I decided to dig in further on how to reproduce this bug in-game, and as it turns out, the problem is with the duplicated ragdolls needing to be faced in a specific direction from where the player faces when entering a level.

So if the duplicated ragdolls are spawned from a different angle, that causes the gizmos and such, to be misaligned. I have NEVER seen anything like this before.

**Comment by ToxicHolyGrenade** [comment]
@penolakushari But on top of that, the rotational gizmos become misaligned even if the precise angle is correct. It blows my mind.

**Comment by ToxicHolyGrenade** [comment]
![hl2 2024-01-15 19-01-26-55](https://github.com/Winded/RagdollMover/assets/83011225/06b0b2a8-2db3-43ca-bf24-463ca6f77bcc)
![hl2 2024-01-15 19-01-38-43](https://github.com/Winded/RagdollMover/assets/83011225/67176226-9117-4e07-9207-3d212f3efae5)
![hl2 2024-01-15 19-01-47-85](https://github.com/Winded/RagdollMover/assets/83011225/fce4befa-f07e-44ef-ae5c-08a83d3fa0c5)

**Comment by ToxicHolyGrenade** [comment]
@penolakushari The other question is, would it be wise to start over and recreate the addon, so long as the source code isn't thrown completely out? Or can we use ChatGPT to overhaul the source code?

**Comment by penolakushari** [comment]
Where's the problem? There's none, as the gizmos seem to be in the localized mode, and position and angle ones use separate localized toggles, and those use angles of the currently selected bone, so "local" to it. Therefore it is more of a thing with the ragdoll you're using having pelvis bone angled like that. To have stuff local to "world coordinates" disable these checkboxes at the top of the tool's C menu.
![image](https://github.com/Winded/RagdollMover/assets/45951611/17da7b43-4f4b-46cc-adec-b5dc901e1f00)
Please do not suggest using chatgpt, I can't really access it anyway and the current update seems to be shaping up well without it.

**Comment by ToxicHolyGrenade** [comment]
Localized? angles and positions? My bad. I'll try this out again, and I will keep you noted if there are other problems.

**Comment by ToxicHolyGrenade** [comment]
> Where's the problem? There's none, as the gizmos seem to be in the localized mode, and position and angle ones use separate localized toggles, and those use angles of the currently selected bone, so "local" to it. Therefore it is more of a thing with the ragdoll you're using having pelvis bone angled like that. To have stuff local to "world coordinates" disable these checkboxes at the top of the tool's C menu. ![image](https://private-user-images.githubusercontent.com/45951611/296871623-17da7b43-4f4b-46cc-adec-b5dc901e1f00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2MDgzMDEsIm5iZiI6MTcwNTYwODAwMSwicGF0aCI6Ii80NTk1MTYxMS8yOTY4NzE2MjMtMTdkYTdiNDMtNGY0Yi00NmNjLWFkZWMtYjVkYzkwMWUxZjAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDIwMDAwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU1ZWQwZGQ4MGZlNjEyNTllZWJjYjRkNDNmYTAxMjU2MTBjMGNlNjdiMTYzNjI4OWU0M2YxY2IzMGYwMDlhOTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.V5BBBlwhd5BvyZ4Kriw3PeXHTg36KSjMmZnaAwOK9uE) Please do not suggest using chatgpt, I can't really access it anyway and the current update seems to be shaping up well without it.

Bad news, I unchecked the localized gizmo options, and I am still getting those misaligned issues. My theory is that somehow, depending on white height or angle I have saved the duplicated ragdolls, results in the gizmos being positioned wrongly. It is mainly the duplications we are talking about.

![hl2 2024-01-18 20-04-58-98](https://github.com/Winded/RagdollMover/assets/83011225/948f6e27-518a-4442-bf74-e8781f601e0c)

![hl2 2024-01-18 20-05-19-40](https://github.com/Winded/RagdollMover/assets/83011225/9a7f6385-6990-47bb-b32d-7c9911552fb0)

**Comment by ToxicHolyGrenade** [comment]
@penolakushari The other theoretical cause, is the stand tool/feature. If the previous issues I have noted aren't what's causing this, perhaps make the ragdoll mover tool more compatible with the stand tool.

**Comment by ToxicHolyGrenade** [comment]
> What you are showing is the gizmos in the world angles mode, so it's working as intended. If you want them to assume angles of your currently selected bone, then enable that stuff back up. If you want to speed up the thing with some binds, you can use some "bind toggle" stuff with ragmover_localang and ragmover_localpos which use 0 or 1 as arguments

Okay, as much as I don’t want to be blunt, but why the heck would you manufacture a feature like this if it is going to bring more hardships than benefits? If anything, world mode should rather be left out of it, or at the very least, have it kept in a more advanced variation of the same tool.

Because while there is no question there may be features that will provide more usefulness, the world mode, from the way its designed, hardly brings anything useful to the table. I’m sorry to break your dream, but that’s the reality of why some, if not all, features can be abusive and unfriendly to the user to the point things will become like a hassle fest. Useful features would be like ones that provide more efficiency, and make things easier. The likes of world mode, has none of that.

**Comment by penolakushari** [comment]
I'm afraid that the only one who makes stuff harder here is you. World/Localized gizmo transforms feature was present in ragdoll mover since it was made by Winded way back with more simpler functionality of mostly moving physics stuff, before I've added all the bells and whistles for manipulating nonphysical bones or effect prop stuff. Whole thing can be just enabled through the localized gizmos check and due to how console variables work, it'll remember the setting between your play sessions, so you may just as well forget about it after enabling, if it's that much of a bother.

**Comment by ToxicHolyGrenade** [comment]
You may have a valid point, but the problem is, the I am not that convinced, because even after I have disabled the Ragdoll Mover Localized gizmos, the gizmos are still misaligned with the duplicated ragdolls, which makes no sense in a bad way. Despite the evidence I gave, you guys do not believe me. I literally disabled the check boxes off, and it still happened. Recently today, I checked the boxes on and off, still results in the same outcome. If anything, if its not the fault of the tool, then it is the fault of the Garry's Mod duplication system, if that's anything to go by.

**Comment by penolakushari** [comment]
The duped thing is still the ragdoll, it's not some animated entity that breaks everything once it gets duped? Perhaps you could upload it to steam workshop as a usual dupe so I may just as well check it on my end, steam and all.

**Comment by ToxicHolyGrenade** [comment]
> The duped thing is still the ragdoll, it's not some animated entity that breaks everything once it gets duped? Perhaps you could upload it to steam workshop as a usual dupe so I may just as well check it on my end, steam and all.

No, not really, because from what I remembered, I never converted the ragdoll into an animation entity at the time I was reworking on Counteractor's appearance. Plus, I also recently did Neo, who is a Spy TF2 Freak, and I saved him to my workshop dupes.

Neo: https://steamcommunity.com/sharedfiles/filedetails/?id=3144270436

Required Addons

https://steamcommunity.com/sharedfiles/filedetails/?id=135491961

https://steamcommunity.com/sharedfiles/filedetails/?id=514736326

Windy Sally Parker/Counteractor: https://steamcommunity.com/sharedfiles/filedetails/?id=2949350786 & https://steamcommunity.com/sharedfiles/filedetails/?id=2949662824

Required Addons

https://steamcommunity.com/sharedfiles/filedetails/?id=2891075442

https://steamcommunity.com/sharedfiles/filedetails/?id=312899896

https://steamcommunity.com/sharedfiles/filedetails/?id=405793043

**Comment by ToxicHolyGrenade** [comment]
Actually, sorry, scratch that, its not just the duplications of ragdolls, it also occurs with ragdolls that are positioned with the Stand Tool. I tried that recently just now. All you have to do, is go on Gmod, spawn a ragdoll, use the stand tool and position the ragdoll in a particular angle, use the ragdoll mover, and see if the problem gets reproduced.

**Comment by penolakushari** [comment]
Nope, couldn't reproduce the issue you are describing. Could you try to record a video of how you turn on and off the options whereas having some bone on the ragdoll selected? It switching orientations should be visible there, as well as you did say that such an issue persists even with all other addons off.

**Comment by ToxicHolyGrenade** [comment]
> Nope, couldn't reproduce the issue you are describing. Could you try to record a video of how you turn on and off the options whereas having some bone on the ragdoll selected? It switching orientations should be visible there, as well as you did say that such an issue persists even with all other addons off.

Are you referring to the comment I had posted about my duplications, or are you referring to the one mentioning the stand tool as of now?

**Comment by penolakushari** [comment]
Happened to go through and dupe, and stand pose stuff, all gizmo angles work as intended in any case. Having some possibility to see what happens on your end would give some insight I'd say.

**Comment by ToxicHolyGrenade** [comment]
As for recording a video, I tried, but the file size is too big unless I optimize it.

**Comment by penolakushari** [comment]
You could upload it to youtube I guess, it generally translates it into its own format that handles the rest. May take a while though

**Comment by ToxicHolyGrenade** [comment]
![hl2 2024-01-23 19-00-38-56](https://github.com/Winded/RagdollMover/assets/83011225/2a364db8-3712-44dc-b452-41b2e8f1b3c9)
![hl2 2024-01-23 19-01-03-00](https://github.com/Winded/RagdollMover/assets/83011225/dc4e0623-4144-414e-9b4a-3d19e55f6f7f)
![hl2 2024-01-23 19-01-09-56](https://github.com/Winded/RagdollMover/assets/83011225/b51d5c29-1818-4b34-a423-01730e046ee7)
![hl2 2024-01-23 19-01-13-02](https://github.com/Winded/RagdollMover/assets/83011225/5a67289a-8026-4c93-b5c3-701d60054709)
![hl2 2024-01-23 19-01-41-03](https://github.com/Winded/RagdollMover/assets/83011225/64d3fb7a-9c75-4255-9d86-cacaf4ae23d4)
![hl2 2024-01-23 19-01-47-79](https://github.com/Winded/RagdollMover/assets/83011225/4863c993-4357-418d-abd1-2972472e026a)

**Comment by ToxicHolyGrenade** [comment]
![hl2 2024-01-23 19-01-57-90](https://github.com/Winded/RagdollMover/assets/83011225/3526ee28-878b-4c3b-b899-48f0116392fc)
![hl2 2024-01-23 19-08-45-41](https://github.com/Winded/RagdollMover/assets/83011225/88b27d53-f15a-42d7-9ec8-f0c4cf68ed4e)
![hl2 2024-01-23 19-08-52-86](https://github.com/Winded/RagdollMover/assets/83011225/022fa383-f384-47b0-8926-97792517347e)
![hl2 2024-01-23 19-09-01-31](https://github.com/Winded/RagdollMover/assets/83011225/5c890517-3563-4f84-b015-62f0800f9b19)

**Comment by ToxicHolyGrenade** [comment]
![hl2 2024-01-23 19-09-06-75](https://github.com/Winded/RagdollMover/assets/83011225/57c7f552-064e-4ca9-8f2d-9f89456efc0b)
![hl2 2024-01-23 19-09-13-35](https://github.com/Winded/RagdollMover/assets/83011225/69bead99-41c1-472f-9afe-9b8c7706e56b)
![hl2 2024-01-23 19-09-18-11](https://github.com/Winded/RagdollMover/assets/83011225/f30dd235-3cad-4733-8cc5-44e03cc750b3)
![hl2 2024-01-23 19-09-38-42](https://github.com/Winded/RagdollMover/assets/83011225/100a1c41-905d-4d14-ac04-b93e5f20f163)

**Comment by ToxicHolyGrenade** [comment]
![hl2 2024-01-23 19-09-40-69](https://github.com/Winded/RagdollMover/assets/83011225/a1871705-5a01-43fc-bf24-9b4ce085485a)
![hl2 2024-01-23 19-09-56-52](https://github.com/Winded/RagdollMover/assets/83011225/3aed0757-6a1b-4910-966c-46e75ae5f70a)

**Comment by penolakushari** [comment]
Alrighty, the 2 checkboxes are for different modes of ragdoll mover, position gizmo for the moving arrows one, and angle gizmo is the discs one. Disabling those puts them in the world angles mode, as I did happen to write before, enabling them makes them align to the currently selected bone. Scaling gizmo doesn't have that as scaling stuff around world axises would be kinda weird.

In simpler terms, use upper checkbox to switch how the arrows are located, one below it for the discs rotation mode.

**Comment by ToxicHolyGrenade** [comment]
I am aware of that, but I checked them on and off, but still the same problem. It is either the problem with Garry's Mod's fundamental errors causing this, or the stand tool. Because you did insist that there is nothing wrong with the ragdoll mover tool probably, then it is the other tools causing it, to which I have mentioned already in this very comment.

**Comment by Thespikedballofdoom** [comment]
What is the misalignment anyway? is it that it is in world space, or that the arrows aren't perfectly straight downward? Many model pelvis bones are oriented at an odd angle by default in their true standing mode.

**Comment by ToxicHolyGrenade** [comment]
> What is the misalignment anyway? is it that it is in world space, as in not in localized mode, or that the arrows aren't perfectly straight downward when in localized mode? Many model pelvis bones are oriented at an odd angle by default in their true standing mode.

Not just the misalignment of the angle and position gizmos, but also the results of the position gizmos not being correctly straight, as shown in one of the images.

**Comment by Thespikedballofdoom** [comment]
Which image. Again if they aren't perfectly downward, that's probably the pelvis being oriented in the default pose

**Comment by ToxicHolyGrenade** [comment]
> Which image. Again if they aren't perfectly downward, that's probably the pelvis being oriented in the default pose

https://private-user-images.githubusercontent.com/83011225/299076073-3526ee28-878b-4c3b-b899-48f0116392fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxODg2NDAsIm5iZiI6MTcwNjE4ODM0MCwicGF0aCI6Ii84MzAxMTIyNS8yOTkwNzYwNzMtMzUyNmVlMjgtODc4Yi00YzNiLWI4OTktNDhmMDExNjM5MmZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDEzMTIyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmYmRlMjk2NDBkYTIyZTFhOGE5ZmZlYTA4YTcyZmIwZWZjY2I4MmVlNTQ0NjlkNGU5Yzc5YzRmMTc5YzcwYjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.np9vWR3I7qD0MdsBSy_FWD7FDQCagATxpXOdTU9B_-8

The pelvis is perfectly fine, but that doesn't change the fact that the gizmos in this image here is misaligned and broken.

**Comment by Thespikedballofdoom** [comment]
It isn't misaligned.
![image](https://github.com/Winded/RagdollMover/assets/43627353/2b3a3a5e-7139-4802-b0c4-42759e4f0f8a)
This is what it looks like for me, how it's supposed to be oriented when localized angles are on.
Check out scout's .qc:
![image](https://github.com/Winded/RagdollMover/assets/43627353/d1820926-5e70-4dca-a886-9da22aac9f74)
It's rotated by default.

**Comment by ToxicHolyGrenade** [comment]
> It isn't misaligned. ![image](https://private-user-images.githubusercontent.com/43627353/299678824-2b3a3a5e-7139-4802-b0c4-42759e4f0f8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxODk1OTgsIm5iZiI6MTcwNjE4OTI5OCwicGF0aCI6Ii80MzYyNzM1My8yOTk2Nzg4MjQtMmIzYTNhNWUtNzEzOS00ODAyLWIwYzQtNDI3NTllNGYwZjhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDEzMjgxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmYTY1MTFkZjM2MjEzZTkwYmEzZGY5ZjgxZTZhNjYzMDJiMjYxYzViNDk2MTgzN2UwODJiMGYxMGQ2MjMwOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.k69XIz49OAf6Qagnp-NwNL9_8gF9zPOwuHD1Reb-FF0) This is what it looks like for me, how it's supposed to be oriented when localized angles are on. Check out scout's .qc: ![image](https://private-user-images.githubusercontent.com/43627353/299680548-d1820926-5e70-4dca-a886-9da22aac9f74.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxODk1OTgsIm5iZiI6MTcwNjE4OTI5OCwicGF0aCI6Ii80MzYyNzM1My8yOTk2ODA1NDgtZDE4MjA5MjYtNWU3MC00ZGNhLWE4ODYtOWRhMjJhYWM5Zjc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDEzMjgxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhMDZlYmNlZTU2MWM4Yzc3NDdjMTg0NTEzOTI4MGZmMWI0YTdiODc0OTYwMDRlOGI4ZGNkYzgzZDc5MmU2MTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.79H8naeNxa8Z_YDQ9afcjuGlpoDKiW1fEacwALEfPGQ) It's rotated by default.

Is there any way we can patch this out, both Garry's Mod and all the other Scout based ragdolls? Should we encourage the addon creators to do the same?

**Comment by penolakushari** [comment]
That's why world orientation is a thing, in case you want to move stuff up or down without caring how bone is rotated, or to rotate a foot in IK rig for example

**Comment by ToxicHolyGrenade** [comment]
> It isn't misaligned. ![image](https://private-user-images.githubusercontent.com/43627353/299678824-2b3a3a5e-7139-4802-b0c4-42759e4f0f8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY0NjQ3NTUsIm5iZiI6MTcwNjQ2NDQ1NSwicGF0aCI6Ii80MzYyNzM1My8yOTk2Nzg4MjQtMmIzYTNhNWUtNzEzOS00ODAyLWIwYzQtNDI3NTllNGYwZjhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI4VDE3NTQxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2ODEyYzRmNTY5Njk5NDNkMjRjZDQzNGFiZDY2YTc3MDA3YzYwODRjNWExMDIxYzA0MmM5YTg0OWE3YmUwZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6OBAO_A8966ngMadmWza2RV1Rsl_kF8nARl0VW9cxCE) This is what it looks like for me, how it's supposed to be oriented when localized angles are on. Check out scout's .qc: ![image](https://private-user-images.githubusercontent.com/43627353/299680548-d1820926-5e70-4dca-a886-9da22aac9f74.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY0NjQ3NTUsIm5iZiI6MTcwNjQ2NDQ1NSwicGF0aCI6Ii80MzYyNzM1My8yOTk2ODA1NDgtZDE4MjA5MjYtNWU3MC00ZGNhLWE4ODYtOWRhMjJhYWM5Zjc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI4VDE3NTQxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxMjQ1MmU1YjRmZjVhNWYwYjk5YWVjM2EzN2VlZjcxZDZkMTIwMzI5MTNkOGIzNDFjMTViNjAyNGY1ZTA2YTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HLvXAcmPYvQ3rsemB9BI_wNfWpGpZOKkSqktHshb-8U) It's rotated by default.

Would it be a good idea to do so?

**Comment by ToxicHolyGrenade** [comment]
> It isn't misaligned. ![image](https://private-user-images.githubusercontent.com/43627353/299678824-2b3a3a5e-7139-4802-b0c4-42759e4f0f8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc2NzY5ODAsIm5iZiI6MTcwNzY3NjY4MCwicGF0aCI6Ii80MzYyNzM1My8yOTk2Nzg4MjQtMmIzYTNhNWUtNzEzOS00ODAyLWIwYzQtNDI3NTllNGYwZjhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjExVDE4MzgwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkZjI0YzdiMmRkOTZiMzdjZDkzMGQ5NGEzY2EyYTA4ZmE2N2VhYjY4OTE2MmVjMjgyNjAxOWE2NjQ4N2MwYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Z_Y64_pKC2o47wOy_XMTY3aLtJ994yBNWFaXpKKIpl8) This is what it looks like for me, how it's supposed to be oriented when localized angles are on. Check out scout's .qc: ![image](https://private-user-images.githubusercontent.com/43627353/299680548-d1820926-5e70-4dca-a886-9da22aac9f74.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc2NzY5ODAsIm5iZiI6MTcwNzY3NjY4MCwicGF0aCI6Ii80MzYyNzM1My8yOTk2ODA1NDgtZDE4MjA5MjYtNWU3MC00ZGNhLWE4ODYtOWRhMjJhYWM5Zjc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjExVDE4MzgwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1NjgyOTZjZGE0YTkyMTBiY2QzMzViZWM2OWE0ZTExMGJkNjUyYmFmMmVjODY0YzVkZTg4ZWM0MWYxNjFmNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4zLT1S8ZfhUwwUbOhEyoa-vVMyaPwGzcJ23wCtD0UhU) It's rotated by default.

If I want to change the pelvis value so it is perfectly straight in order to fix the misalignment problem, then what is the value exactly to do so?

**Comment by ToxicHolyGrenade** [comment]
> That's why world orientation is a thing, in case you want to move stuff up or down without caring how bone is rotated, or to rotate a foot in IK rig for example

Okay, while I understand it might sound too far fetched, what about implementing a .qc bone orientation editor/correction feature into the Ragdoll Mover, so the gizmos won't look out of place anymore on most ragdolls like the Scout?
